### PR TITLE
Comment and Post models new fields

### PR DESCRIPTION
**Pull request by Magnico**
Changes made:
- Field created_at added to Comment and Post model
- Fields comments_count, safe and img added to Post model
- Tests updated for new fields

### CODE REVIEW

File: blog/models.py [hunk 1]
```diff
@@ -1,19 +1,44 @@
...
+def get_upload_path(instance, filename):
+    return 'uploads/images/{}/{}/{}/{}'.format(instance.created_at.year, instance.created_at.month, instance.created_at.day, filename)
 
 # Create your models here.
 class Post(models.Model):
     title = models.CharField(max_length=100,null=False,blank=False)
     body = models.CharField(max_length=255,null=False,blank=False)
     author = models.ForeignKey(User,on_delete=models.CASCADE, null=False)
+    created_at = models.DateTimeField(auto_now_add=True,editable=False)
+    img = models.ImageField(upload_to=get_upload_path,null=True,blank=True)
```
Comment (ejariza-evowill) [inline blog/models.py]: This is not necessary, as ImageField (or more specifically, its parent FileField) supports [strftime formatting.](https://docs.djangoproject.com/en/4.1/ref/models/fields/#filefield)

File: blog/models.py [hunk 2]
```diff
@@ -1,19 +1,44 @@
...
+        self.img.delete(save=False)
+        super().delete(*args, **kwargs)
+
+    def save(self, *args, **kwargs):
+        if self.pk:
+            #Look for the original image
+            original = Post.objects.get(pk=self.pk)
+            #If the original image is different from the new one, delete the original image
+            if original.img and original.img != self.img:
+                original.img.delete(save=False)
```
Comment (ejariza-evowill) [inline blog/models.py:31]: Why do you consider it's necessary to do this?

File: blog/tests.py [hunk 3]
```diff
@@ -243,6 +333,8 @@ def test_post_patch(self):
         self.assertEqual(post.title, response.data['title'])
         self.assertEqual(post.body, response.data['body'])
         self.assertEqual(post.author.username, response.data['author'])
+        self.assertEqual(post.safe, response.data['safe'])
+        self.assertEqual(post.img.name, response.data['img'] if response.data['img'] else '')
```
Comment (ejariza-evowill) [inline blog/tests.py]: Why the conditional here? shouldn't the test expect always the same results?

**Review by ejariza-evowill — CHANGES_REQUESTED** [review]

File: blog/models.py (hunk 2 above)
Comment (Magnico) [reply blog/models.py:31]: when I was testing the functionality of uploading images through the django administrator, I saw that when deleting or changing an image in the post the image itself remained saved in the corresponding directory.

File: blog/tests.py (hunk 3 above)
Comment (Magnico) [reply blog/tests.py]: Happens that when there is no image attached, the response have `None` in the `response.data['img']` and the `post.img.name` would be `''` so I use the conditional for that case.

File: blog/models.py (hunk 2 above)
Comment (ejariza-evowill) [reply blog/models.py:31]: Oh, ok, I didn't specify to delete images, but it still is a sensible decision, so it's ok.

File: blog/tests.py (hunk 3 above)
Comment (ejariza-evowill) [reply blog/tests.py]: But if in the test the logic is well isolated it should happen only one of both cases.

Comment (Magnico) [reply blog/tests.py]: Yes, i understand now what you mean. I am correcting the test.

**Comment by Magnico** [comment]
Corrections made.

File: blog/tests.py [hunk 4]
```diff
@@ -125,7 +125,7 @@ def test_post_img_update(self):
         post.save()
 
         #check if the image have the correct path
-        expected_path = f'uploads/images/{post.created_at.year}/{post.created_at.month}/{post.created_at.day}/test_update_2.jpg'
+        expected_path = f'uploads/images/{post.created_at.strftime("%Y/%m/%d")}/test_update_2.jpg'
```
Comment (ejariza-evowill) [inline blog/tests.py:128]: Good addition, strftime ins a great help here.

**Review by ejariza-evowill — APPROVED** [review]

**Pull request merged**